src/main/java/com/derongan/minecraft/deeperworld/movement/transition/SectionTransition.kt

```diff
@@ -0,0 +1,29 @@
+package com.derongan.minecraft.deeperworld.movement.transition
+
+import com.derongan.minecraft.deeperworld.event.PlayerAscendEvent
+import com.derongan.minecraft.deeperworld.event.PlayerChangeSectionEvent
+import com.derongan.minecraft.deeperworld.event.PlayerDescendEvent
+import com.derongan.minecraft.deeperworld.world.section.Section
+import org.bukkit.Location
+import org.bukkit.entity.Player
+
+data class SectionTransition(
+    val from: Location,
+    val to: Location,
+    val fromSection: Section?,
+    val toSection: Section?,
+    val kind: TransitionKind
+)
+
+enum class TransitionKind {
+    ASCEND,
+    DESCEND
+}
+
+internal fun SectionTransition.toEvent(player: Player): PlayerChangeSectionEvent {
+    return if (this.kind == TransitionKind.ASCEND) {
+        PlayerAscendEvent(player, fromSection!!, toSection!!)
```

Comment:
Either we don't specify to/from sections as nullable or there is a good reason they are and it's handled properly.